usdoj-crt/beta-ada
# JavaScript Refactor to use ES6 modules

Changes:
All JS files: 
1. Removed the Sprockets convention of requiring in modules using `//require "module"`
2. Replaced with ES modules. Note that in some scripts the revealing module pattern was used along with IIFEs, these have been removed in favor of `import` and `export`.
3. A few `vars` were replaced with `let` and `const`. This is not done universally as I decided to focus on the `import/export` statements for this PR. 
4. Note that all functions are the default export unless there are multiple items being exported from the module.
5. Change to `scripts.html` - Made sure the pagination script is imported from the `/dist` folder.

To Test:
Basically we just need to make sure that all of the JS still functions as before.
1. Check that the 'skip to main content' and 'skip to page navigation' buttons still work. Press 'tab' when first arriving on a page to get those buttons to appear, then press 'enter'.
2. Confirm that the accordions open properly. Confirm that they close other accordions that are already open. The accordions look like this: <img width="646" alt="Screen Shot 2022-03-02 at 9 48 27 AM" src="https://user-images.githubusercontent.com/14644234/156384995-74023e15-405b-4998-a412-0cc1c28896e4.png">
3. Confirm that the top 'How you can tell you're on a .gov website' banner opens.
4. Confirm the print button on the Topics pages triggers page printing
5. Confirm that the search feature works
6. Confirm that the search pagination works - should display a total number of results, with 20 results per page, and a list of buttons corresponding to the total number of pages below.
7. Confirm that the urls in the search results wrap on small screen sizes.
